src/map/script.cpp

```diff
@@ -19705,7 +19705,7 @@ BUILDIN_FUNC(bg_get_data)
  *------------------------------------------*/
 //Returns an Instance ID
 //Checks NPC first, then if player is attached we check
-unsigned short script_instancegetid(struct script_state* st)
+unsigned short script_instancegetid(struct script_state* st, bool use_mode, enum instance_mode mode)
```

Comment:
Instead of two parameters make it default IM_NONE and check if it is set to something else.

Reply:
Good idea !

Reply:
The 'auto-detect' of the instance ID of the player for `instance_enter`, `instance_destroy` etc.. commands who are using `script_instancegetid` will still be broken if the player owns 2 instances (for example IM_PARTY and IM_GUILD instances types).

For me the 'auto-detect' lead to bug, I would prefer to remove it and make, instead, IM_PARTY by default in `script_instancegetid`. The instance ID will always be required for the instances commands for instances types other than IM_PARTY (=add explanations in the docs).

Sample of the idea:
```
izlude,100,100,0	script	test instance	56,{
	// If player owns IM_PARTY and IM_GUILD, by default the command will destroy IM_PARTY instance.
	instance_destroy;
	end;
}

izlude,101,100,0	script	test instance 2	56,{
	// ID always required to destroy IM_GUILD instance.
	instance_destroy instance_id(IM_GUILD);
	end;
}

izlude,101,100,0	script	test instance 3	56,{
	// If player owns IM_PARTY and IM_GUILD, by default the command warp the player in IM_PARTY instance.
	instance_enter("Endless Tower");
	end;
}

izlude,101,100,0	script	test instance 4	56,{
	// ID always required to warp in IM_GUILD instance.
	instance_enter( "Endless Tower", -1, -1, getcharid(0), instance_id(IM_GUILD) );
	end;
}
```

Reply:
Yes good idea